migrator/migrator.go

```diff
@@ -515,13 +515,22 @@ func (m Migrator) MigrateColumn(value interface{}, field *schema.Field, columnTy
 		}
 
 		// check precision
-		if precision, _, ok := columnType.DecimalSize(); ok && int64(field.Precision) != precision {
-			if regexp.MustCompile(fmt.Sprintf("[^0-9]%d[^0-9]", field.Precision)).MatchString(m.DataTypeOf(field)) {
-				alterColumn = true
+		if realDataType == "decimal" {
+			precision, scale, ok := columnType.DecimalSize()
+			if ok {
+				if !strings.HasPrefix(fullDataType, fmt.Sprintf("decimal(%d,%d)", precision, scale)) &&
+					!strings.HasPrefix(fullDataType, fmt.Sprintf("decimal(%d)", precision)) {
```

Comment:
Should we check for prefix or full string match here?

Reply:
fullDataType maybe like: ``` decimal(10,1) not null```